### PR TITLE
[US029] Damaging player

### PR DESCRIPTION
**Pull request by krzysiekprzekwas**
Implemented colision between bullets and players and added health to player

### CODE REVIEW

File: GameServer/Game/GameEngine.cs [hunk 1]
```diff
@@ -79,10 +83,31 @@ private void Tick(object state)
...
             // Send game state for every connected client
              _hubContext.Clients.All.SendAsync("updateGameState", GameState.Instance);
         }
 
+        private void ApplyDamage()
+        {
+            var bulletHitIds = new List<int>();
+            foreach (var bullet in _gameState.Bullets)
+            {
+                foreach (var player in _gameState.Players)
```
Comment (tomejzen) [inline GameServer/Game/GameEngine.cs]: Maybe we can create new method called Intersection.CheckAnyIntersectionWithPlayers

File: GameServer/Models/Player.cs [hunk 2]
```diff
@@ -30,7 +31,7 @@ public Player()
         {
             Id = GameState.Instance.GeneratePlayerUniqueId();
             Keys = new List<KeyEnum>();
-
+            Health = 100;
```
Comment (tomejzen) [inline GameServer/Models/Player.cs]: MAGIC NUMBER DETECTED!!!

File: GameServer/wwwroot/index.html [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
     <span id="fpsLabel" class="measurementLabel">0.00 - FPS</span>
     <span id="pingLabel" class="measurementLabel">0 - Ping</span>
+    <span id="healthLabel" class="measurementLabel">100/100</span>
```
Comment (tomejzen) [inline GameServer/wwwroot/index.html]: It can be empty before player receives data about his health.

File: GameServer/wwwroot/js/world.js [hunk 4]
```diff
@@ -52,11 +52,15 @@ let world = {
...
                 radius: player.radius,
-                angle: player.angle
+                angle: player.angle,
+                health: player.health
             };
 
-            if (playerObject.id === world.playerId)
+            if (playerObject.id === world.playerId) {
                 world.myPlayer = playerObject;
+                $('#healthLabel').text(world.myPlayer.health+"/100");
```
Comment (tomejzen) [inline GameServer/wwwroot/js/world.js]: I would love to see here:

`{world.myPlayer.health}/{world.myPlayer.maxhealth}`

But it is only my suggestion. Have a nice evening! ;-)

File: GameServer/Models/Player.cs (hunk 2 above)
Comment (krzysiekprzekwas) [reply GameServer/Models/Player.cs]: Yeah, so You could stick it deeply up in Your Good-Practise Ass

**Comment by tomejzen** [comment]
No response to: "Maybe we can create new method called Intersection.CheckAnyIntersectionWithPlayers"

Anyway great job!

**Pull request merged**